src/k2/module/k23si/TxnManager.cpp

```diff
@@ -64,7 +64,7 @@ seastar::future<> TxnManager::start(const String& collectionName, dto::Timestamp
         K2LOG_D(log::skvsvr, "txn manager check hb");
         _hbTask = _hbTask.then([this] {
             // refresh the clock
-            auto now = CachedSteadyClock::now(true);
+            auto now = Clock::now();
```

Comment:
Why this fix the issue?

Reply:
The problem was that the cachedclock is only updated on the hb timer callback. If a txn starts before the first HB callback is executed, its expiry date is very stale. The fix now is to refresh the cached clock on TxnManager::start